docs/decisions/0001-launch-complete-course.rst

```diff
@@ -0,0 +1,421 @@
+paths used to render the course content, we can achieve this by wrapping
+the routes with a function that will change available routes for
+the LTI launch user. This approach will require the use of the LTI plugin
+and the learning MFE.
+
+Workflow
+--------
+
+1. In the component *TabPage* we will validate if an LTI launch is in
+   the request.
+2. If an LTI launch is requested, the function *isLtiPluginEnabled* is called.
+3. *isLtiPluginEnabled* will call an API endpoint on our plugin that
+   will validate if the plugin is enabled (This could also evaluate the
+   LMS or site settings if possible from the learning MFE).
+4. If the API responds with a 200 code (or the settings
+   OLTTP_ENABLE_LTI_TOOL on the LMS is true), *isLtiPluginEnabled* is
+   set to true and the function *LtiProfileExists* is called.
+5. *LtiProfileExists* will validate that the user in the requests
+   contains an *LTIProfile* model instance and that this instance ID is
+   equal to the LTIProfile ID on the request.
+6. If *isLtiPluginEnabled* and *LtiProfileExists* are false, the result
+   of *TabPage* is unchanged.
+7. If *isLtiPluginEnabled* and *LtiProfileExists* are true:
+
+   1. the function *isAllowedLtiPath* will be called to validate that
+      the requested path is allowed for LTI launches.
+   2. If *isAllowedLtiPath* is false, we return no component to the
+      user.
+   3. If *isAllowedLtiPath* is true:
+
+      1. We will remove the *Header* and *Footer* component from the
+         result on *TabPage*.
+
+Hiding unwanted content
+-----------------------
+
+If *isAllowedLtiPath* is true, we could send an *isLtiLaunch* parameter
+to *LoadedTabPage* and its children component with this we could:
+
+- On the *Course* component, we will hide any component that could take
+  the user outside course content (Example: *SidebarTrigger*
+  component).
+- On the *OutlineTab* component, we could apply the same (Example:
+  *UpgradeNotification* component).
+
+Workflow Diagram
+----------------
+
+.. image:: https://mermaid.ink/img/pako:eNptkstuwjAQRX9l5DX8QBZFiEehYoFKukqyGOKBWHXs1B4LocC_12kITaVmZcXnnrFv0orSShKJODtsKkiXuYH4zNtc7NItaAymrMDRVyDPYA2keNzjmWa5uD9QmE5fbmvUnm6wzB77oDwEg5rJkSw6bkynLkR40XYjGh3OygAZPGqSs4d18cc6ftdnV-2HJwcVeoiSvbMnpeNQ4xlNSYNl9Y9lNbKs262HBrkC1Npefsevx8HX7J04uHjGuuHr0EAxJnvdZgCHEi6KKxsYNoQynhaNhLW1sZNHeNOFYZsdKO4ov2O16wtv0GFNEQS2T5vr5aWtG2vIsC-etW5_RG_ZvCscykpp6WiMxrAPmgsxETW5GpWM37ztwrngimrKRRKXEt1nLnJzjxwGtoerKUXC8XYTERqJTEuF8VepRXLqyrl_A1FtvKc?type=png)](https://mermaid.live/edit#pako:eNptkstuwjAQRX9l5DX8QBZFiEehYoFKukqyGOKBWHXs1B4LocC_12kITaVmZcXnnrFv0orSShKJODtsKkiXuYH4zNtc7NItaAymrMDRVyDPYA2keNzjmWa5uD9QmE5fbmvUnm6wzB77oDwEg5rJkSw6bkynLkR40XYjGh3OygAZPGqSs4d18cc6ftdnV-2HJwcVeoiSvbMnpeNQ4xlNSYNl9Y9lNbKs262HBrkC1Npefsevx8HX7J04uHjGuuHr0EAxJnvdZgCHEi6KKxsYNoQynhaNhLW1sZNHeNOFYZsdKO4ov2O16wtv0GFNEQS2T5vr5aWtG2vIsC-etW5_RG_ZvCscykpp6WiMxrAPmgsxETW5GpWM37ztwrngimrKRRKXEt1nLnJzjxwGtoerKUXC8XYTERqJTEuF8VepRXLqyrl_A1FtvKc
+   :width: 729
```

Comment:
No change required, but did you know you can put Mermaid markup [directly in GitHub these days](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/)? Just thought it'd be more convenient for you folks in the future.

Reply:
Thanks!, I didn't know, its the first time I use Mermaid on Github so I wasn't aware of the feature, will consider it next time I use a diagram on Github.